internal/addon/manifests/charts/mcoa/charts/metrics/templates/deployment.yaml

```diff
@@ -0,0 +1,90 @@
+{{- if .Values.enabled }}
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: metrics-addon-agent
+  namespace: {{ .Values.addonInstallNamespace }}
+  labels:
+    app.kubernetes.io/part-of: multicluster-observability-addon
+    app.kubernetes.io/component: metrics-agent
+    app.kubernetes.io/instance: metrics-agent
+    app.kubernetes.io/name: metrics-addon-agent
+    app.kubernetes.io/version: 2.48.1
```

Comment:
nit: You can make this label block much easier to maintain with helm by creating a`_helpers.tpl`. Similar to how we did for logging. We should also have some consistency on what labels we have. This however can be a follow-up. This can also be done for selectors.

Reply:
Also for the version, you can set it as a value and then when you want to update the version you only need to make 1 change. And no need to make it customizable in the addon if there is no need for it

Reply:
I don't see where it's done for logging. The current helpers there do not contain labels helpers: https://github.com/rhobs/multicluster-observability-addon/blob/ef986fd7172230976ddc1623c6b10b45243e4957/internal/addon/manifests/charts/mcoa/charts/logging/templates/_helpers.tpl